src/transformers/modeling_utils.py

```diff
@@ -2629,7 +2629,11 @@ def _fix_key(key):
         # This is not ideal in terms of memory, but if we don't do that not, we can't initialize them in the next step
         if low_cpu_mem_usage:
             for key in missing_keys:
-                if key.startswith(prefix):
+                if key in list(model_state_dict.keys()):
+                    key = key
+                elif ".".join([prefix] + key.split(".")) in list(model_state_dict.keys()):
+                    key = ".".join([prefix] + key.split("."))
+                elif ".".join(key.split(".")[1:]) in list(model_state_dict.keys()):
```

Comment:
```suggestion
                elif key.startswith(prefix) and ".".join(key.split(".")[1:]) in list(model_state_dict.keys()):
```
We still need to test if it starts with the prefix here, otherwise we might catch false positives here.

Reply:
yes, I will add this in the next commit.